<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:macro name="imu">

    <ros2_control name="imu" type="sensor">
      <hardware>
          <plugin>imu_sensor_hardware/IMUSensorHardware</plugin>
          <!-- <param name="use_fake_mode">${use_fake_mode}</param> -->
      </hardware>
      <sensor name="imu">
        <state_interface name="orientation.x"/>
        <state_interface name="orientation.y"/>
        <state_interface name="orientation.z"/>
        <state_interface name="orientation.w"/>
        <state_interface name="angular_velocity.x"/>
        <state_interface name="angular_velocity.y"/>
        <state_interface name="angular_velocity.z"/>
        <state_interface name="linear_acceleration.x"/>
        <state_interface name="linear_acceleration.y"/>
        <state_interface name="linear_acceleration.z"/>
      </sensor>

    </ros2_control>

  </xacro:macro>

</robot>
